api/query_engine.go

```diff
@@ -38,41 +38,178 @@ func (a archives) Len() int           { return len(a) }
 func (a archives) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
 func (a archives) Less(i, j int) bool { return a[i].interval < a[j].interval }
 
-// updates the requests with all details for fetching, making sure all metrics are in the same, optimal interval
-// luckily, all metrics still use the same aggSettings, making this a bit simpler
-// note: it is assumed that all requests have the same from, to and maxdatapoints!
-// this function ignores the TTL values. it is assumed that you've set sensible TTL's
-func alignRequests(reqs []models.Req, aggSettings mdata.AggSettings) ([]models.Req, error) {
+// represents a numbered data "archive", i.e. the raw one, or an aggregated series
```

Comment:
Why are there 2 almost identical data structures (archive and narchive).  Surely you can just add the position variable (i) to archive and remove narchive.  Just because it is defined doesnt mean it has to be used.